advocacy_docs/edb-postgres-ai/migration-etl/data-migration-service/getting_started/preparing_db/preparing_postgres_source_databases.mdx

```diff
@@ -0,0 +1,198 @@
+
+#### Config validation script
+
+The EDB DMS Reader installation (packaged as `cdcreader`) comes with a helper script that validates the Postgres configuration and helps you identify any issues. After you configure the database, we recommend running the script and ensuring all checks passed.
+
+Run the script without arguments to print the usage:
+
+```shell
+/opt/cdcreader/postgresConfigValidation.sh
+```
+
+### Create new roles and grant acccess for CDC migration
+
+First, create a new role for CDC migration with `LOGIN` and `REPLICATION` abilities granted:
+
+```sql
+CREATE ROLE <MIGRATION_ROLE> WITH REPLICATION LOGIN PASSWORD '<MIGRATION_ROLE_PASSWORD>';
+```
+
+`<MIGRATION_ROLE>` needs to own the source tables to autocreate Postgres publications. Because the source tables are already owned by another role, you create a role/user that can act as the new owner and grant the specified replication group role to both the current table owner and to `<MIGRATION_ROLE>`:
+
+```sql
+CREATE ROLE <REPLICATION_GROUP>;
+GRANT <REPLICATION_GROUP> TO <MIGRATION_ROLE>;
+GRANT <REPLICATION_GROUP> TO <ORIGINAL_OWNER>;
+ALTER TABLE <TABLE_NAME> OWNER TO <REPLICATION_GROUP>
+```
+
+Where:
+
+ - `<MIGRATION_ROLE>` is the name of the Postgres role or user to use for CDC migration database access.
+ - `<ORIGINAL_OWNER>` is the original production owner of the table.
+ - `<REPLICATION_GROUP>` is the name of a role used to own the source tables to migrate for publication autocreation.
+
+### Grant usage of the source schemas to the CDC migration role
+
+To select tables in a schema, the user must grant usage on the schema in addition to granting `SELECT` on the tables of the schema. This is necessary for all source schemas containing tables to be migrated.
+
+You can grant usage on a schema using the following command:
+
+```sql
+GRANT USAGE ON SCHEMA <DB_SCHEMA> TO <MIGRATION_ROLE>;
+```
+
+### Grant `SELECT` on source tables to the CDC migration role
+
+The new `<MIGRATION_ROLE>` now has schema access, but still needs `SELECT` access to the source tables of those schemas. You can grant access across a schema
+or for each table.
+
+For an entire schema's tables, use this command:
+
+```sql
+ALTER DEFAULT PRIVILEGES IN SCHEMA <DB_SCHEMA> GRANT SELECT ON TABLES to <MIGRATION_ROLE>
+```
+
+For each table, use:
+
+```sql
+GRANT SELECT ON <TABLE_NAME> TO <MIGRATION_ROLE>
+```
+
+Where:
+ - `<DB_SCHEMA>` is the database schema name for the tables to migrate.
+ - `<MIGRATION_ROLE>` is the name of the Postgres role or user to use for CDC migration database access.
+ - `<TABLE_NAME>` is the name of a table to migrate.
+
+### Grant `CREATE` privileges on the source database to the CDC migration role
+
+Since the `pgoutput` plugin is being used, the migration user needs to be granted `CREATE` priveleges on the source database in order for Debezium to be able to create publications. The 'CREATE' privelege can be granted using the following command:
```

Comment:
Should 'CREATE' in the following sentence be stylized as a keyword (i.e., `CREATE`) like the other occurrences:
> The 'CREATE' privelege can be granted using the following command: